File: front-end/components/RiderRegister.js
import { View, Text, TextInput, Alert, TouchableOpacity } from "react-native";
import { useState, useEffect } from "react";
import AsyncStorage from "@react-native-async-storage/async-storage";
import { myStyle } from "../styles/mystyle";
import { useNavigation } from "@react-navigation/native";

export function RiderRegister() {
  const [personId, setPersonId] = useState("");
  const [riderFirstname, setRiderFirstname] = useState("");
  const [riderLastname, setRiderLastname] = useState("");
  const [brandVehicle, setBrandVehicle] = useState("");
  const [modelVehicle, setModelVehicle] = useState("");
  const [license, setLicense] = useState("");
  const [riderLicense, setRiderLicense] = useState("");
  const [riderLocation, setRiderLocation] = useState("");
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const [email, setEmail] = useState("");
  const [riderLicenseImage, setRiderLicenseImage] = useState("");
  const [errors, setErrors] = useState({});

  const handleSaveData = async () => {
    if(!validate()){
      return;
    }

    if (
      !personId ||
      !riderFirstname ||
      !riderLastname ||
      !brandVehicle ||
      !modelVehicle ||
      !license || 
      personId.length < 13
    ) {
      return;
    }
    await AsyncStorage.setItem(
      "setRiderRegister",
      JSON.stringify({
        personId,
        riderFirstname,
        riderLastname,
        brandVehicle,
        modelVehicle,
        license,
      })
    );
    navigation.navigate("RiderRegister2");
  };

  const navigation = useNavigation();

  const validate = () => {
    let valid = true;
    let tempError = {};

    if (!personId) {
      tempError.personId = "กรุณากรอกเลขบัตรประชาชน";
      let valid = false;
    } else if (personId.length < 13 || personId.length > 13) {
      tempError.personId = "กรุณากรอกเลขบัตรประชนให้ครบ13หลัก";
      let valid = false;
    }
    if (!riderFirstname) {
      tempError.riderFirstname = "กรุณากรอกชื่อจริง";
      let valid = false;
    }
    if (!riderLastname) {
      tempError.riderLastname = "กรุณากรอกนามสกุล";
      let valid = false;
    }
    if (!brandVehicle) {
      tempError.brandVehicle = "กรุณากรอกผู้ผลิตรถมอเตอร์ไซค์";
      let valid = false;
    }
    if (!modelVehicle) {
      tempError.modelVehicle = "กรุณากรอกรุ่นของรถมอเตอร์ไซค์";
      let valid = false;
    }
    if (!license) {
      tempError.license = "กรุณากรอกทะเบียนรถ";
      let valid = false;
    }

    setErrors(tempError);
    return valid;
  };

  return (
    <View style={{ backgroundColor: "#fff", flex: 1 }}>
      
      <Text
        style={{
          color: "#307A59",
          fontWeight: '900',
          fontSize: 30,
          fontFamily: "inter",
          marginTop: 28,
          marginLeft: 50,
        }}
      >
        สมัครใช้งาน {"\n"}
        สำหรับวินมอเตอร์ไซค์
      </Text>

      <View style={myStyle.mainreg}>
        <Text style={{ fontWeight: "bold", fontSize: 18, marginTop: 25 }}>
          ชื่อจริง
        </Text>
        <TextInput
          style={myStyle.inputreg}
          value={riderFirstname}
          onChangeText={setRiderFirstname}
        />
        <Text style={{ color: "red" }}>{errors.riderFirstname}</Text>

        <Text style={{ fontWeight: "bold", fontSize: 18 }}>นามสกุล</Text>
        <TextInput
          style={myStyle.inputreg}
          value={riderLastname}
          onChangeText={setRiderLastname}
        />
        <Text style={{ color: "red" }}>{errors.riderLastname}</Text>

        <Text style={{ fontWeight: "bold", fontSize: 18 }}>
          ผู้ผลิตรถมอเตอร์ไซค์
        </Text>
        <TextInput
          style={myStyle.inputreg}
          value={brandVehicle}
          onChangeText={setBrandVehicle}
        />
        <Text style={{ color: "red" }}>{errors.brandVehicle}</Text>

        <Text style={{ fontWeight: "bold", fontSize: 18 }}>
          รุ่นและสีของรถมอเตอร์ไซค์
        </Text>
        <TextInput
          style={myStyle.inputreg}
          value={modelVehicle}
          onChangeText={setModelVehicle}
        />
        <Text style={{ color: "red" }}>{errors.modelVehicle}</Text>

        <Text style={{ fontWeight: "bold", fontSize: 18 }}>ทะเบียนรถ</Text>
        <TextInput
          style={myStyle.inputreg}
          value={license}
          onChangeText={setLicense}
        />
        <Text style={{ color: "red" }}>{errors.license}</Text>

        <Text style={{ fontWeight: "bold", fontSize: 18 }}>เลขบัตรประชาชน</Text>
        <TextInput
          style={myStyle.inputreg}
          value={personId}
          onChangeText={setPersonId}
        />
        <Text style={{ color: "red" }}>{errors.personId}</Text>

        <TouchableOpacity 
          style={{
            backgroundColor: "#307A59",
            paddingVertical: 9,
            borderRadius: 10,
            alignItems: "center",
            marginVertical: 10,
            cursor: "pointer",
            width: 130,
            marginTop: 18,
            marginLeft: 180,
          }} 
          onPress={handleSaveData}>
          <Text style={myStyle.buttonLogin}>ถัดไป</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
}

File: back-end/src/main/java/dev/kittpas/greenwin/Entity/RiderRegister.java
package dev.kittpas.greenwin.Entity;

import org.hibernate.annotations.CreationTimestamp;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "riderregister")
public class RiderRegister {

    @Id
    @Column(name = "personId", length = 13)
    private String personId;

    @CreationTimestamp
    @Column(name = "registerTime")
    private String registerTime;

    @Column(name = "riderFirstname")
    private String riderFirstname;

    @Column(name = "riderLastname")
    private String riderLastname;

    @Column(name = "brandVehicle")
    private String brandVehicle;

    @Column(name = "modelVehicle")
    private String modelVehicle;

    @Column(name = "license")
    private String license;

    @Column(name = "riderLicense")
    private String riderLicense;

    @Column(name = "riderLocation")
    private String riderLocation;

    @Column(name = "username")
    private String username;

    @Column(name = "password")
    private String password;

    @Column(name = "email")
    private String email;

    @Column(name = "tel")
    private String tel;

    public RiderRegister() {

    }

    public String getPersonId() {
        return personId;
    }

    public void setPersonId(String personId) {
        this.personId = personId;
    }

    public String getRiderFirstname() {
        return riderFirstname;
    }

    public void setRiderFirstname(String riderFirstname) {
        this.riderFirstname = riderFirstname;
    }

    public String getRiderLastname() {
        return riderLastname;
    }

    public void setRiderLastname(String riderLastname) {
        this.riderLastname = riderLastname;
    }

    public String getBrandVehicle() {
        return brandVehicle;
    }

    public void setBrandVehicle(String brandVehicle) {
        this.brandVehicle = brandVehicle;
    }

    public String getModelVehicle() {
        return modelVehicle;
    }

    public void setModelVehicle(String modelVehicle) {
        this.modelVehicle = modelVehicle;
    }

    public String getLicense() {
        return license;
    }

    public void setLicense(String license) {
        this.license = license;
    }

    public String getRiderLicense() {
        return riderLicense;
    }

    public void setRiderLicense(String riderLicense) {
        this.riderLicense = riderLicense;
    }

    public String getRiderLocation() {
        return riderLocation;
    }

    public void setRiderLocation(String riderLocation) {
        this.riderLocation = riderLocation;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getTel() {
        return tel;
    }

    public void setTel(String tel) {
        this.tel = tel;
    }

        public String getRegisterTime() {
        return registerTime;
    }

    public void setRegisterTime(String registerTime) {
        this.registerTime = registerTime;
    }

    public RiderRegister(String personId, String riderFirstname, String riderLastname, String brandVehicle,
            String modelVehicle, String license, String riderLicense, String riderLocation, String username,
            String password, String email,  String tel, String registerTime) {
        this.personId = personId;
        this.riderFirstname = riderFirstname;
        this.riderLastname = riderLastname;
        this.brandVehicle = brandVehicle;
        this.modelVehicle = modelVehicle;
        this.license = license;
        this.riderLicense = riderLicense;
        this.riderLocation = riderLocation;
        this.username = username;
        this.password = password;
        this.email = email;
        this.tel = tel;
        this.registerTime = registerTime;
    }



}
